src/manageiq_client/api.py

```diff
@@ -616,11 +616,33 @@ def __call__(self, *args, **kwargs):
             self.api.response = action_response
         return outcome
 
+    def _get_entity_from_href(self, result):
+        href_result = result['href']
+        href_match = None
+
+        # If the "href" value in result doesn't match the current collection,
+        # try to find the collection that the "href" refers to.
+        if self.collection._href not in href_result:
+            href_match = re.match(r"(http[^?]+)/([a-z_-]+)", href_result)
+
+        if href_match:
```

Comment:
@mkoura I know not everyone agrees with multiple function exits but I think that it would be much simpler if we reverse the logic a little bit:

```python
def _get_entity_from_href(self, result):
    result_href = result['href']

    if self.collection._href.startswith(href_result):
        return Entity(self.collection, result, incomplete=True)

    ...
```

so we would spare this extra condition.

Reply:
Ok, I don't have problem with that.